Sources/Optimizer/py_stochastic_reconfiguration.cc

```diff
@@ -3,31 +3,59 @@
 #include <pybind11/eigen.h>
 #include <pybind11/pybind11.h>
 
+#include "Utils/exceptions.hpp"
+#include "Utils/pybind_helpers.hpp"
 #include "common_types.hpp"
 
+namespace netket {
+
 namespace py = pybind11;
 
-namespace netket {
-namespace detail {
-namespace {
-void AddSR(py::module m) {
-  py::class_<SR>(m, "SR")
-      .def(py::init<double, bool, bool, bool>(), py::arg{"diag_shift"} = 0.01,
-           py::arg{"use_iterative"} = false, py::arg{"use_cholesky"} = true,
-           py::arg{"is_holomorphic"} = true)
-      .def(
-          "compute_update",
-          [](SR& self, Eigen::Ref<const MatrixXcd> Oks,
-             Eigen::Ref<const VectorXcd> grad,
-             Eigen::Ref<VectorXcd> out) { self.ComputeUpdate(Oks, grad, out); },
-          py::arg{"Oks"}.noconvert(), py::arg{"grad"}.noconvert(),
-          py::arg{"out"}.noconvert());
-}
-}  // namespace
-}  // namespace detail
+void AddSR(py::module& m) {
+  py::class_<SR>(m, "SR", "Performs stochastic reconfiguration (SR) updates")
+      .def(py::init([](const std::string& solver_name, double diag_shift,
+                       bool use_iterative, bool is_holomorphic) {
+             const auto solver = SR::SolverFromString(solver_name);
+             NETKET_CHECK(solver.has_value(), InvalidInputError,
+                          "Invalid LSQ solver \"" << solver_name
+                                                  << "\" specified for SR");
+             return SR(solver.value(), diag_shift, use_iterative,
+                       is_holomorphic);
+           }),
+           py::arg("lsq_solver") = "LLT", py::arg("diag_shift") = 0.01,
+           py::arg("use_iterative") = false, py::arg("is_holomorphic") = true)
+      .def("compute_update", &SR::ComputeUpdate, py::arg("Oks").noconvert(),
+           py::arg("grad").noconvert(), py::arg("out").noconvert(),
+           R"EOF(
+            Solves the SR flow equation for the parameter update ẋ.
+
+            The SR update is computed by solving the linear equation
+               Sẋ = f
+            where S is the covariance matrix of the partial derivatives
+            O_i(v_j) = ∂/∂x_i log Ψ(v_j) and f is a generalized force (the loss
+            gradient).
 
-void AddSR(PyObject* m) {
-  detail::AddSR(py::module{py::reinterpret_borrow<py::object>(m)});
+            Args:
+                Oks: The matrix 𝕆 of centered log-derivatives,
+                   𝕆_ij = O_i(v_j) - ⟨O_i⟩.
+                grad: The vector of forces f.
+                out: Output array for the update ẋ.
+          )EOF")
+      .def_property("store_rank_enabled", &SR::StoreRankEnabled,
```

Comment:
Why not simply `store_rank`?

Reply:
`store_rank` is more of a function name than a property to me. The property is _whether storing the rank is enabled_.